### PR TITLE
bump version

### PR DESCRIPTION
**Pull request by guolinke**

### CODE REVIEW

**Comment by guolinke** [comment]
@jameslamb could you update R's version accordingly? you can create a new PR if needed.

**Comment by jameslamb** [comment]
We don't necessarily know that the next release will be 3.0.1 instead of 3.1.0, right?

What do you think about the convention where the development version is like 3.0.0.9999 (newer than 3.0.0, older than any future release we do with a 3-part number), and then we change to 3.0.1 or 3.1.0 based on what is new?

If any new features are added between now and the next release, then that release should be 3.1.0. If it is only maintenance tasks, 3.0.1

**Comment by guolinke** [comment]
I like the proposal. With that, we can distinguish the dev and release versions.

File: .appveyor.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: 3.0.0.{build}
+version: 3.0.1.{build}
```
Comment (guolinke) [inline .appveyor.yml:1]:
```suggestion
version: 3.0.0.99.{build}
```

File: VERSION.txt [hunk 2]
```diff
@@ -1 +1 @@
-3.0.0
+3.0.1
```
Comment (guolinke) [inline VERSION.txt:1]:
```suggestion
3.0.0.99
```

**Comment by guolinke** [comment]
After merge this, we can start to merge the blocked PRs about new features.

**Comment by jameslamb** [comment]
I'll update the R version right now and make a PR into this one

**Comment by jameslamb** [comment]
I can also upload the R artifacts to the 3.0.0 release manually tonight

**Comment by jameslamb** [comment]
created #3345

**Review by StrikerRUS — APPROVED** [review]

**Pull request merged**